src/Nethermind/Nethermind.Evm/Tracing/GethStyle/Custom/Native/Prestate/NativePrestateTracerAccount.cs

```diff
@@ -0,0 +1,20 @@
+// SPDX-FileCopyrightText: 2024 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System.Collections.Generic;
+using System.Text.Json.Serialization;
+using Nethermind.Int256;
+
+namespace Nethermind.Evm.Tracing.GethStyle.Custom.Native.Prestate;
+
+[JsonConverter(typeof(NativePrestateTracerAccountConverter))]
+public class NativePrestateTracerAccount
+{
+    public UInt256 Balance { get; set; }
+
+    public ulong? Nonce { get; set; }
+
+    public byte[]? Code { get; set; }
+
+    public Dictionary<string, string>? Storage { get; set; }
```

Comment:
maybe better to keep it UInt256->UInt256 and only convert it to string when writing to JSON?

Reply:
I looked into making these changes, however using UInt256 types as dictionary keys isn't currently supported out of the box with the existing JSON serialization framework. I could look into adding making the changes to add this support, however I feel like there would be minimal performance improvements that would come from this since the storage entries only need to be converted once from bytes to string and never need to be deserialized or updated.

Please let me know if you feel strongly about this or if I'm missing something in regards to the potential performance improvements from these changes.

Reply:
I think we did it here `[UInt256Converter.cs](https://github.com/NethermindEth/nethermind/pull/5530/files#diff-7a1f7b16dc0b6a9798bc370212cd256db09859c79b7b5812e62958b6e8d30477)`

feel free to take that code.

Reply:
I spent a lot of today debugging a different issue I was seeing with storage lookup but was just able to get the UInt256 converter working with Oleg's changes. However, the formatting for the UInt256 hex values is off now due to lack of zero padding. The expected format is a hex value with all 32 bytes that is zero padded on the left. Do you think it's worth adding a `ZeroPaddedHex` enum to `NumberConversion` that can be used to format the value with padded zeroes? If so, should I add support for displaying with the new enum to `ulong` and `long` as well?

Also if you have any better ideas about getting this formatting correct that doesn't require a new enum please let me know as well.

Reply:
I pushed up a commit with the `WriteAsPropertyName` updates from the PR that you linked, I left out the `Read/ReadAsPropertyName` updates for now since they're not necessary for my changes